### PR TITLE
fix error overlay sometimes now showing

### PR DESCRIPTION
**Pull request by ForsakenHarmony**

### CODE REVIEW

**Comment by ForsakenHarmony** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3331** <a href="https://app.graphite.dev/github/pr/vercel/turbo/3331" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/vercel/turbo/3331?utm_source=stack-comment).

File: crates/next-core/js/src/dev/hmr-client.ts [hunk 1]
```diff
@@ -116,6 +116,15 @@ function aggregateUpdates(
   const key = resourceKey(msg.resource);
   const aggregated = chunksWithUpdates.get(key);
 
+  if (msg.type === "issues" && aggregated == null && hasIssues) {
+    // add an empty record to make sure we don't call `onBuildOk`
+    chunksWithUpdates.set(key, {
```
Comment (ForsakenHarmony) [inline crates/next-core/js/src/dev/hmr-client.ts:121]: We check if there are entries in the map below, so even if we have no updates we insert an empty record to make sure it doesn't think everything is fine

File: crates/next-core/js/src/dev/hmr-client.ts [hunk 2]
```diff
@@ -116,6 +116,15 @@ function aggregateUpdates(
   const key = resourceKey(msg.resource);
   const aggregated = chunksWithUpdates.get(key);
 
+  if (msg.type === "issues" && aggregated == null && hasIssues) {
```
Comment (jridgewell) [inline crates/next-core/js/src/dev/hmr-client.ts:119]: Nit: Looks like this can be merged with the if statements below?

**Review by jridgewell — APPROVED** [review]

**Comment by wbinnssmith** [comment]
Any chance this fixes WEB-439?

File: crates/next-core/js/src/dev/hmr-client.ts (hunk 2 above)
Comment (ForsakenHarmony) [reply crates/next-core/js/src/dev/hmr-client.ts:119]: could be, but imo makes it harder to read

**Comment by ForsakenHarmony** [comment]
@wbinnssmith it might

**Pull request merged**